### ISSUE TITLE
ML eğitim pipeline'ın izlenebilirliği ve sonuç denetimi için log/artifakt iyileştirmesi

### ISSUE THREAD

**Issue by SefaGH** [SefaGH/bearish-alpha-bot#309]
# Amaç
Eğitim pipeline'ındaki ölçülebilirlik ve operasyonel denetim eksiklerini minimize etmek. Model eğitim sonuçlarının, metriklerinin ve eğitim loglarının denetlenebilir, analiz edilebilir, arşivlenebilir şekilde saklanmasını sağlamak.

# Sorunun Özeti
- Model eğitim süreci tamamlanıyor ve model dosyaları başarılı şekilde arşivleniyor.
- Eğitim logları ve ilerleme metrikleri (epoch/loss, val_loss vs.) ise stdout'ta kayboluyor, artifakt olarak kaydedilmiyor.
- ccxt ve urllib3 gibi kütüphanelerin aşırı DEBUG logları, GitHub Actions stdout limitini dolduruyor. Önemli eğitim logları ve hata analizleri gözden kaçabiliyor.

# Eylem Planı

1. **Gürültülü log seviyelerinin düşürülmesi** 
   - Kodun başında, şu şekilde seviye azalt:
     ```python
     import logging
     logging.getLogger("ccxt.base.exchange").setLevel(logging.WARNING)
     logging.getLogger("urllib3").setLevel(logging.WARNING)
     ```
   - Kendi projenin logger'ı: default INFO, debugda DEBUG.

2. **Eğitim metriklerinin dosyaya zorunlu ve düzenli kaydedilmesi**
   - Eğitim döngüsünde:
     ```python
     logging.info("Epoch %d/%d | loss=%.6f", epoch, num_epochs, loss)
     ```
   - `logs/metrics.csv` veya `logs/metrics.json` dosyasına: epoch, train_loss, val_loss, lr, süre.
   - Eğitim bitiminde log ve metrik dosyasının path'ini logla (`logger.info("Saved metrics: logs/metrics.csv")`).

3. **Log ve metrik dosyalarını model artifaktlarıyla birlikte arşivleme**
   - CI (GitHub Actions) workflow'unda şu dosyaları `actions/upload-artifact@v4` ile arşivle:
     - `logs/training.log`
     - `logs/metrics.csv`
     - model dosyaları
   - Dosya adları: örneğin `training-log-${{ github.sha }}`
   - Retention (silme süresi) parametresi ekle (`retention-days: 14` vs.).
   - Arşiv öncesi `ls -lah data/models/` ile model içeriklerini göster.

4. **Sembol filtresi ile stdout azaltımı**
   - Sadece hedef sembollerle eğitim (
BTC, ETH...)
   - ccxt'de `symbols` filtresi ile sadece ihtiyaç duyulan enstrümanları çek, geri kalan işlenmesin.
   - Eğitim başında sembol listesini logla.

5. **GPU kullanılabilirliğini raporla**
   - `torch.cuda.is_available()` ve aktif cihazın adını eğitim başında tek logda göster:
     ```python
     logger.info(f"CUDA Available: {torch.cuda.is_available()} | Device: {torch.cuda.get_device_name(0) if torch.cuda.is_available() else 'CPU' }")
     ```

6. **Model arşiv adını zenginleştir (SHA/run_no)**
   - Arşiv isimleri: `trained-models-${{ github.sha }}-${{ github.run_number }}`
   - Model arşivi yüklenmeden önce `ls -lah data/models/` çıktısı alınmalı.

# Kod Referansları
- `scripts/train_all_models.py` (eğitim metrik ve log kaydı, sembol filtresi)
- `.github/workflows/train-models.yml` (artifact upload adımları, retention, dosya adlandırma)
- `src/ml/model_trainer.py` ve `src/ml/price_predictor.py` (eğitim loop'unda `metrics.csv` oluşturulması)
- Logger ayarı: projenin `setup_logger` fonksiyonu

# Sonuç
Bu adımlar uygulanarak ML eğitim pipeline'ı tam operasyonel denetlenebilir hale gelir; hata/root cause, model performansı, overfitting ve eğitim hataları dakikalar içinde saptanabilir ve analiz edilebilir olur.
